Tools/flatpak/setup-rpkgs/R/functions.R

```diff
@@ -0,0 +1,775 @@
+    installGitHubRecords(githubRecords, c(normalizePath(lib), libpaths[length(libpaths)]))
+
+  }
+
+}
+
+downloadFile <- function(url, destdir) {
+  mkdir(destdir)
+  destfile <- file.path(destdir, basename(url))
+  download.file(url = url, destfile = destfile)
+  return(destfile)
+}
+
+downloadRenv <- function(destdir) {
+  # TODO: don't hardcode the renv version? maybe use 1 older than the current release so the url always works?
+  downloadFile("https://cran.r-project.org/src/contrib/renv_0.13.2.tar.gz", destdir)
+}
+
+downloadRemotes <- function(destdir) {
+  downloadFile("https://cran.r-project.org/src/contrib/remotes_2.4.0.tar.gz", destdir)
+}
+
+
+copyRfiles <- function(dirs) {
+  file.copy(from = list.files("R", pattern = "*\\.R$", full.names = TRUE), to = dirs["r-helpers"], overwrite = TRUE)
+}
+
+createTarArchive <- function(dirs, outputPath = "archives/flatpak_archive.tar.gz", compression = c("fast", "none", "best"),
+                             verbose = TRUE, update = FALSE) {
+
+  # TODO: update does not work
+
+  if (update && !identical(compression[1L], "none")) {
+    warning("can only update when compression is 'none'! Not updating instead.", immediate. = TRUE)
+    update <- FALSE
+  }
+
+  if (!is.numeric(compression)) {
+    compression <- match.arg(compression)
+    compression <- switch(compression,
+      "none" = "",
+      "best" = "-I 'gzip -9'",
+      "fast" = "-I 'gzip -1'"
+    )
+  } else if (!is.numeric(compression) || (is.numeric(compression) && !(compression >= 1 && compression <= 9))) {
+    stop("compression must be character ('best' or 'fast') or numeric (1-9)")
+  }
+
+  dirsForArchive <- dirs["flatpak-dir"]
+  # dirsForArchive <- c(dirs["flatpak-dir"], file.path(dirs["renv-root"], "source", "github"))
+
+  # files <- file.path("..", basename(getwd()), makePathRelative(dirsForArchive))
+  files <- makePathRelative(dirsForArchive, base = dirname(dirs["root"]))
+
+  mkdir(dirname(outputPath))
+
+  creatArchive <- sprintf(
+    "tar %s -%s%sf %s %s",
+    compression,
+    if (verbose) "v" else "",
+    if (update)  "u" else "c",
+    outputPath,
+    paste(files, collapse = " ")
+  )
+  cat('running: ', creatArchive, '\n')
+  system(creatArchive)
+
+  sha256 <- strsplit(system2("sha256sum", outputPath, stdout = TRUE), " ", fixed = TRUE)[[1]][1]
+  cat("\nsha256sum\n")
+  cat(sha256, '\n')
+
+  rfile <- makePathRelative(file.path(dirs["r-helpers"], "flatpakRegenerateRenvStructure.R"), base = dirs["root"])
+  cat(sprintf("command for flatpak\nR --vanilla --file=%s\n", rfile))
+
+  return(c(
+    "tar-file" = outputPath,
+    "sha256"   = sha256,
+    "r-file"   = rfile
+  ))
+}
+
+uploadTarArchive <- function(archivePath = "archives/flatpak_archive.tar.gz") {
+
+  if (!file.exists(archivePath))
+    stop("Archive does not exist")
+
+  archivePath <- normalizePath(archivePath)
+
+  cmd <- sprintf("scp %s jonathonlove@static.jasp-stats.org:static.jasp-stats.org/flatpak_archive.tar.gz", archivePath)
```

Comment:
Aah yeah as mentioned above: maybe add like the commit from jasp-desktop to the name or something? So we can have multiple versions on the server.